zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/BaseDataAgent.scala

```diff
@@ -0,0 +1,116 @@
+package edu.uci.ics.cloudberry.zion.actor
+
+import akka.actor.Props
+import akka.pattern.pipe
+import edu.uci.ics.cloudberry.zion.common.Config
+import edu.uci.ics.cloudberry.zion.model.datastore.{IDataConn, IQLGenerator}
+import edu.uci.ics.cloudberry.zion.model.schema._
+import org.joda.time.{DateTime, Duration}
+import play.api.libs.json._
+
+import scala.concurrent.ExecutionContext
+
+class BaseDataAgent(override val dbName: String,
```

Comment:
The name of "BaseDataAgent" is a little bit confusing with "AbstractDataSetAgent", since base and abstract are sometimes both used for "parent/super" class in OOP. I would suggest using a different name for "BaseDataAgent", such as "DataAgent"?